core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -208,23 +201,14 @@ public static Catalog loadCatalog(
    */
   public static Catalog buildIcebergCatalog(String name, Map<String, String> options, Configuration conf) {
     String catalogImpl = options.get(CatalogProperties.CATALOG_IMPL);
+    String catalogType = options.get(CatalogProperties.CATALOG_TYPE);
+    Preconditions.checkArgument(catalogImpl == null || catalogType == null,
+        "Cannot create catalog %s, both type and catalog-impl are set: type=%s, catalog-impl=%s",
```

Comment:
Oh you're right about `type` being an existing Spark config 😅. I had been working too much in Flink when I first looked at this. My bad.

In regards to changing the Flink catalog, the need to use `type = iceberg` is something that I believe is mandated by Flink. So I don't think it will be possible to change that aspect.